src/repositories/request-repository.ts

```diff
@@ -170,7 +180,7 @@ export class RequestRepository {
     const keys = Object.keys(request).filter((key) => key !== 'id') // all keys except ID
     const [{ id }] = await connection
       .table(TABLE_NAME)
-      .insert(request, ['id'])
+      .insert(request.toDB(), ['id'])
```

Comment:
hmm... this scares me a little.  What happens if a developer fails to call `.toDB()` - will something clearly error?  Or will it silently insert the wrong data?

Reply:
It will silently insert the wrong data, and it's okay. Someone has to explicitly convert runtime format to a database one. I am open to suggestions though.

I am generally not happy about runtime <-> db conversion. See no good solution for now, but it does not mean the search is over.

Reply:
ooph yeah that's a bummer.  Is there maybe a way we could configure the database to yell at us if the wrong type is inserted?  Like, if the date fields are inserted as Date objects, what do they show up as in the db?  Are they coerced to strings somehow or do they go in as numbers?  Can we make it so the db transaction would fail in that case?

It's weird that this seems to maybe be one place where TypeOrm was better than knex - TypeOrm has a way to specify custom serializer/deserializers for types, but my googling isn't finding anything equivalent for knex 😭